### ISSUE TITLE
Unexpected behaviour when using a promise condition with wait()

### ISSUE THREAD

**Issue by tejohnso** [SeleniumHQ/selenium#464]
This is either a documentation error, an implementation error, or a misunderstanding on my part.

Documentation: 
```
wait(condition, opt_timeout, opt_message)

In the event a condition returns a promise, the polling loop will
wait for it to be resolved and use the resolved value for whether
the condition has been satisfied. 
```
There are two points of interest here.  First, we wait until the promise is resolved.  Second, we use the **resolved value** to determine whether the condition is satisfied.

The behaviour as implemented doesn't seem to match.  The **resolved value** isn't considered at all.

For example, when I use wait (promise.fullfilled(false)) the condition is satisfied and flow continues past the wait, even though the resolved value is false.

The code in question is here
https://github.com/SeleniumHQ/selenium/blob/0a9cc30d576c7290dcd54ccfd54aff27058192ff/javascript/webdriver/promise.js#L1558

I would expect something like this before `fulfill(value)`
``` javascript
if (!value) {
   return reject(Error(opt_message +
  ": promise resolved to a falsy value"));
}
```

**Comment by jleyba** [comment]
Addressed by #337?

**Comment by jleyba** [comment]
It has been a month with no response; closing ticket.
